### ISSUE TITLE
Roadmap to 1.0

### ISSUE THREAD

**Issue by Ben-G** [ReSwift/ReSwift#91]
Had this at the back of my mind for quite a while. We should discuss the roadmap for getting ReSwift to 1.0, since we've been really close for a while.

On my list I have:

- Solid Example App (working on one right now; the more the merrier)
- Updated Docs (a good portion of documentation is slightly out of date with recent API changes)
- Nice to have: best practice / new API for async operations (can be post 1.0 as well)

I'll be traveling a fair bit in March, so **my goal is to ship 1.0 in April**. 

My plan was to push ReSwift (core) to 1.0 and potentially the router as well and keep ReSwift-Recorder in an experimental state for now.

Thoughts, ideas, concerns?

Let's use this thread to discuss other things that might be important.

**Comment by RubenSandwich** [comment]
I think for version 1.0 documentation is hugely important. Especially because many iOS developers have not used Redux before. I plan on using this library for a few internal projects pretty soon so I'll see if I can swing some time to help with the documentation.

**Comment by Ben-G** [comment]
Agree! I'm starting some restructuring of the docs on a separate branch: https://github.com/ReSwift/ReSwift/tree/benji/1.0-docs

Along with the recording of the talk and the example app we're building, this should be a good starting point.

@RubenSandwich if you identify areas that are especially unclear it would be awesome if you could share them. Any form of contribution to the docs is also very welcome. I would imagine that something like a small step-by-step tutorial could be very useful, too!

**Comment by Aleksion** [comment]
I'll also arise from the dead and contribute to the docs. April is clearing up on my end and I would love to get back into this.

Great work Ben!

**Comment by Ben-G** [comment]
Most work is currently done on the Getting Started Guide. You can follow along here: https://github.com/ReSwift/ReSwift/blob/benji/1.0-docs/Docs/Getting%20Started%20Guide.md

Any thoughts so far?

**Comment by agentk** [comment]
It's looking good. A lot easier to follow with it all in the one guide.

**Comment by agentk** [comment]
Couple of notes that should probably be included. Mainly for newcomers to highlight gotchas that they may not be obvious at first sight. 

State: it's normally good practice to denormalize state objects.

Reducers: need to be fast, avoid making any changes unless a matching action is passed in. E.g. Don't sort and return an array inside a reducer on every invocation. Ideally something like array sorting would be pushed to the Subscribers / ViewModels.

**Comment by TheSooth** [comment]
@Ben-G Hi, if u need help with example app, I can help :smile:

**Comment by Ben-G** [comment]
@TheSooth help would definitely be appreciated! Have you already started working on a project?

**Comment by TheSooth** [comment]
@Ben-G nope, before starting I would to discuss with u, what exactly we want to show in example app :)

**Comment by jlampa** [comment]
How about implementing an authentication process? Every app needs it, so it
would be a natural first step to get to know the framework starting with
the example instead of from scratch.



-- 
Pozdrawiam,

JL

**Comment by RubenSandwich** [comment]
Might I suggest two different example projects, one more of a tutorial example and one more of an actual real world app example. This way someone with no familiarity with with a redux like architecture can look at the tutorial example and then have the foundation to understand the non-trivial example. For the tutorial example I was thinking something in the vein of the [Elm Architecture Tutorial](https://github.com/evancz/elm-architecture-tutorial); a counter, a list of counters, then a dynamic list of counters and finishing on a network image loader.

**Comment by AlexLittlejohn** [comment]
I have an example app that you might be able to use: [Weight](https://github.com/AlexLittlejohn/Weight). It's a weight tracker I made to teach myself the framework. I'm not sure if it follows best practices and so on but I would be willing to clean up/change it however necessary.

**Comment by avocade** [comment]
Docs is definitely a prio, but hammering home the need for immutability in the Store/Reducers and fleshing out the best practices for async actions (which is really the only tricky part imho) is important as well. I'll see what I can contribute once I start using it a bit more heavily in the next weeks.

**Comment by vitkon** [comment]
I have a quick example of facebook login using reSwift:
https://github.com/vitkon/redux-swift-example

**Comment by Ben-G** [comment]
Thanks a lot for sharing your example projects! I was traveling for the last 2 weeks but hopefully this weekend I should find some time to take a look at them and provide feedback.

Really appreciate your participation!

**Comment by vitkon** [comment]
Getting started part needs to be in more detail. It took me some time to figure out how to wire it up.
